projects/ngx-colors/src/lib/directives/ngx-colors-trigger.directive.ts

```diff
@@ -135,14 +136,23 @@ export class NgxColorsTriggerDirective
     this.onChangeCallback(value);
   }
 
-  writeValue(value) {
+  writeValue(value, previewColor = "") {
     if (value !== this.color) {
       if (this.format) {
         let format = formats.indexOf(this.format.toLowerCase());
         value = this.service.stringToFormat(value, format);
       }
       this.color = value;
-      this.change.emit(value);
+
+      let isCmyk = false;
+      if( value && value.startsWith('cmyk')) {
+        isCmyk = true;
+        if( !previewColor ) {
+          previewColor = this.service.stringToFormat(value, ColorFormats.RGBA);
+        }
+      }
+
+      this.change.emit( isCmyk ? previewColor : value);
```

Comment:
why are you emiting the previewValue? why you need to pass it to writeValue ?

Reply:
When this event is triggered, the value can be: "#XXXXXX" or "rgb(XX, XX, XX)" or  "hsl(291, 47%, 51%)" and this text is is used on inline css preview color, like "style: backgound-color:  rgb(XX, XX, XX) " and this is a valid formar for browser render it.

In this case, if the color is CMYK, the css inline style will be "style: backgound-color:  cmyk(XX, XX, XX, XX) " And the browser can not renderise the color. So, if the color is cmyk, i send the previewColor, because the preview color is in Hex format, like "#XXXXXX"